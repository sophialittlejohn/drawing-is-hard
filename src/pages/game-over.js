import React from "react";
import { useLocation, useHistory } from "react-router-dom";
import { Container, Header, Grid, Card, Button, Icon } from "semantic-ui-react";

const gameOverCopy = {
  low: "Hmmm, you can probably do better than that!",
  mid: "Not so shabby. Bet you can beat that in the next round!",
  high: "Wow, impressive! You deserve a margarita at the beach 🍸🌴",
};

export const GameOver = () => {
  const { replace } = useHistory();
  const { state } = useLocation();

  const score = state ? state.score : 0;

  const copy =
    score >= 9
      ? gameOverCopy.high
      : score >= 5
      ? gameOverCopy.mid
      : gameOverCopy.low;

  return (
    <Container style={{ paddingTop: "10%" }} text>
      <Grid>
        <Grid.Row>
          <Grid.Column>
            <Card fluid color="blue">
              <Card.Content>
                <Header as="h1">Drawing is hard - Final score {score}</Header>
              </Card.Content>
            </Card>
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <Card fluid raised>
              <Card.Content style={{ fontSize: "20px" }}>
                <p>GAME OVER!</p>
                <p>
                  You scored <strong>{score}</strong> points.
                </p>
                <p>{copy}</p>
              </Card.Content>
            </Card>
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column width="14">
            <Button
              onClick={() => replace("/game")}
              size="massive"
              color="blue"
              fluid
            >
              Play again
            </Button>
          </Grid.Column>
          <Grid.Column width="two">
            <Button
              onClick={() => replace("/")}
              size="massive"
              fluid
              icon
              basic
              color="blue"
              aria-label="home"
            >
              <Icon name="home" />
            </Button>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Container>
  );
};
