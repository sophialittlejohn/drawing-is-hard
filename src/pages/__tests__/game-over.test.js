import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react";
import { GameOver } from "../game-over";

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
  useLocation: () => ({
    state: { score: 10 },
  }),
  useHistory: () => ({
    replace: jest.fn(),
  }),
}));

it("should take a snapshot", () => {
  const { asFragment } = render(<GameOver />);

  expect(asFragment(<GameOver />)).toMatchSnapshot();
});

it("the final score should be in the DOM", () => {
  const { getByText } = render(<GameOver />);

  expect(getByText("10")).toBeInTheDocument();
});

it("Play again button should be in the DOM", () => {
  const { getByText } = render(<GameOver />);

  expect(getByText(/Play again/i)).toBeInTheDocument();
  fireEvent.click(getByText(/Play again/i));
});

it("Home button should be in the DOM", () => {
  const { getByRole } = render(<GameOver />);

  expect(getByRole("button", { name: /home/i })).toBeInTheDocument();
  fireEvent.click(getByRole("button", { name: /home/i }));
});
