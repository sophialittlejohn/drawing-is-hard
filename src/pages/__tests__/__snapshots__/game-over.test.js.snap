// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui text container"
    style="padding-top: 10%;"
  >
    <div
      class="ui grid"
    >
      <div
        class="row"
      >
        <div
          class="column"
        >
          <div
            class="ui blue fluid card"
          >
            <div
              class="content"
            >
              <h1
                class="ui header"
              >
                Drawing is hard - Final score 10
              </h1>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column"
        >
          <div
            class="ui fluid raised card"
          >
            <div
              class="content"
              style="font-size: 20px;"
            >
              <p>
                GAME OVER!
              </p>
              <p>
                You scored 
                <strong>
                  10
                </strong>
                 points.
              </p>
              <p>
                Wow, impressive! You deserve a margarita at the beach üç∏üå¥
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="fourteen wide column"
        >
          <button
            class="ui blue massive fluid button"
          >
            Play again
          </button>
        </div>
        <div
          class="two wide column"
        >
          <button
            aria-label="home"
            class="ui blue massive basic fluid icon button"
          >
            <i
              aria-hidden="true"
              class="home icon"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
